tests/MongoDB.Driver.Examples/ChangeStreamExamples.cs

```diff
@@ -122,5 +126,77 @@ public void ChangeStreamExample3()
                 next.FullDocument.Should().Be(documents[1]);
             }
         }
+
+        [Fact]
+        public void Changestream_Example_4()
+        {
+            RequireServer.Check().Supports(Feature.AggregateAddFields);
+
+            var client = DriverTestConfiguration.Client;
+            var database = client.GetDatabase("ChangeStreamExamples");
+            database.DropCollection("inventory");
+            var inventoryCollection = database.GetCollection<BsonDocument>("inventory");
+
+            var cancelationTokenSource = new CancellationTokenSource();
+            try
+            {
+                BsonDocument document = new BsonDocument("username", "alice");
+
+                Task.Run(() =>
+                {
+                    while (!cancelationTokenSource.IsCancellationRequested)
+                    {
+                        Thread.Sleep(TimeSpan.FromMilliseconds(100));
+                        document["_id"] = Guid.NewGuid();
```

Comment:
Nope, I've added this line because when I bumped the delay value down (fromm 500 to 100), there was an error from db side (I guess):
`A bulk write operation resulted in one or more errors.
  E11000 duplicate key error collection: ChangeStreamExamples.inventory index: _id_ dup key: { : ObjectId('5bfc110fb8545c1b903fe587') }`.
I assume that it's some kind of optimization process and server is trying to combine two insertOne request to the single one after step when the _id were generated (maybe I'm wrong).

Anyway, when I explicitly specify _id value, this error is not occurred and test work fine.

Reply:
By the way I fount the following similar problem:

Reply:
Note: in case Task.Run will throw any exception by any reason until enumerator.MoveNext() will be passed, MoveNext method will infinitely wait for the first change. As I see, previous examples have the same problem. Should we consider this potential problem as a bug?

Reply:
So, "note behavior" as well as above mentioned duplicate exception can make test execution as infinitely. Should we handle this somehow?

Reply:
> Let's use "ObjectId.GenerateNewId" instead of "Guid.NewGuid" just because MongoDB in general doesn't use GUIDs very often.

I will do this

Reply:
I see now that letting the driver generate the _id results in duplicate key exceptions.

I was confused because I didn't think there would ever be more than one document inserted (that's all that's needed) but the way the test is written any number of documents *might* be inserted, hence the duplicate key exception.